### ISSUE TITLE
document redis version for sails adaptor to work

### ISSUE THREAD

**Issue by mattlo** [balderdashy/sails#3104]
connect-redis versions 1.5.0 and 2.0.0 break on sails version 0.11.0. Only `connect-redis@1.4.5` works.

Currently, sails@0.11/express@3.x injects `connect`, which is not what the newer versions of `connect-redis` expects (`connect.session` is what needs to be injected). Please provide documentation to specify which official adaptor versions to use to avoid other dev/ops from trying to figure this out.

Relates to #2379

**Comment by apowers313** [comment]
+1

After switching my session store from mongodb to redis through the connect-redis@1.4.5 adapter, sessions no longer work. I suspect that connect-redis + express@3.x doesn't actually work, since the connect-redis README points out that using express@3.x also requires using express-session.

Let me know what I can do to help debug.

**Comment by apowers313** [comment]
@mattlo did you ever find a way to make redis work with sails?

**Comment by mattlo** [comment]
@apowers313 this should be working out of the box with 1.4.5 adaptor. Here are my configs:
*config/session.js*

```js
module.exports.session = {
    secret: '***',
    cookie: {
        maxAge: 60 * 60 * 24 * 7
    },
    adapter: 'redis',
    host: process.env.REDIS_HOST || '127.0.0.1',
    port: process.env.REDIS_PORT || 6379,
    ttl: 60 * 60 * 24 * 7,
    db: null,
    pass: process.env.REDIS_PASS || null,
    prefix: 'sess:'
};
```

**Comment by mattlo** [comment]
As for the concern on express version compatibility, I believe 1.4.5 was built for express 3.x. The session module mentioned is when using the newer version of redis-connect on express 3.x.

Refer to documentation on the 1.4.5 tree https://github.com/tj/connect-redis/tree/d542d885c92bc6c3776e4840defc33456b311f9d

**Comment by apowers313** [comment]
@mattlo Not sure how you figured that out, but that did the trick. Thanks.

**Comment by mattlo** [comment]
:+1:

**Comment by tjwebb** [comment]
When you use connect-redis 1.5.0, sails should output a warning that says to use 1.4.5

**Comment by sailsbot** [comment]
Thanks for posting, @mattlo.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
